### PR TITLE
feat: added case-users endpoint mappings with new models

### PR DESCRIPTION
**Pull request by pd-dudkiewicz**
### JIRA link

https://tools.hmcts.net/jira/browse/CMC-1124

### Change description

Adding case-user endpoints mappings with new models to allow sharing the case.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by timja — COMMENTED** [review]
tests? https://github.com/hmcts/ccd-client/tree/master/src/functionalTest/java/uk/gov/hmcts/reform/ccd/client

they aren't run automatically as far as I remember but they are used to validate the client works properly

**Comment by linusnorton** [comment]
@timja the majority of changes are from @pd-dudkiewicz but I had to make some changes to the lombok and wiremock side to fix the tests

File: src/main/java/uk/gov/hmcts/reform/ccd/client/model/CallbackRequest.java [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
 @Data
 @Builder
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/ccd/client/model/CallbackRequest.java:13]: what's the reason for these? previously this client was built using a builder pattern why is the API being compromised by adding a 0 args constructor?

File: src/main/java/uk/gov/hmcts/reform/ccd/client/model/AboutToStartOrSubmitCallbackResponse.java [hunk 2]
```diff
@@ -10,6 +11,7 @@
 
 @Data
 @Builder
+@AllArgsConstructor
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/ccd/client/model/AboutToStartOrSubmitCallbackResponse.java:14]: Is this needed? I think https://projectlombok.org/features/experimental/Jacksonized should be added

File: src/main/java/uk/gov/hmcts/reform/ccd/client/CaseAssignmentApi.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import uk.gov.hmcts.reform.ccd.client.model.CaseAssignmentUserRolesRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseAssignmentUserRolesResource;
+import uk.gov.hmcts.reform.ccd.client.model.CaseAssignmentUserRolesResponse;
+
+import java.util.List;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+@FeignClient(
+        name = "core-case-data-api",
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/ccd/client/CaseAssignmentApi.java:21]: does this need a unique name?

**Review by timja — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Pull request merged**